### PR TITLE
chore: fix create-basket cli

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/client/basket/tx.go [hunk 1]
```diff
@@ -113,9 +113,9 @@ Flags:
 				return err
 			}
 
-			if minStartDateString == "" && startDateWindowString == "" {
+			if minStartDateString == "" && startDateWindow == 0 {
```
Comment (aleem1314) [inline x/ecocredit/client/basket/tx.go:116]: is `DateCriteria` actually required? We are allowing empty `DateCriteria` in [ValidateBasic](https://github.com/regen-network/regen-ledger/blob/release/v2.2.x/x/ecocredit/basket/msg_create.go#L88-L90).

Comment (robert-zaremba) [reply x/ecocredit/client/basket/tx.go:116]: not, it's not required.

**Review by robert-zaremba — APPROVED** [review]
pre-approving -- need to update the date crieria related flags, both of them can be empty, and the DateCriteria object in that case must be `nil`

File: x/ecocredit/client/basket/tx.go [hunk 2]
```diff
@@ -113,9 +113,9 @@ Flags:
...
 				return fmt.Errorf("either min-start-date or start-date-window required")
-			} else if minStartDateString != "" && startDateWindowString != "" {
+			} else if minStartDateString != "" && startDateWindow != 0 {
 				return fmt.Errorf("both min-start-date and start-date-window cannot be set")
```
Comment (robert-zaremba) [inline x/ecocredit/client/basket/tx.go:119]:
```suggestion
				return fmt.Errorf("both %s and %s cannot be set", FlagStartDateWindow, FlagMinimumStartDate)
```

**Comment by robert-zaremba** [comment]
oh, the github tab didn't get refreshed - seams all is already solved. Thanks.

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**